### PR TITLE
chore: Schematics - create CDS schematics

### LINKED ISSUE #12046: Create CDS schematics
Title.

### PR DESCRIPTION
**Pull request by znikola**
closes #12046

### CODE REVIEW

File: projects/schematics/src/shared/utils/lib-utils.ts [hunk 1]
```diff
@@ -86,7 +86,12 @@ export interface FeatureConfig {
   /**
    * An optional custom configuration to provide to the generated module.
    */
-  customConfig?: { import: Import[]; content: string };
+  customConfig?: CustomConfig | CustomConfig[];
```
Comment (marlass) [inline projects/schematics/src/shared/utils/lib-utils.ts:89]: To make it simpler let's only allow for arrays

File: tools/schematics/testing.ts [hunk 2]
```diff
@@ -128,7 +129,7 @@ function testAllSchematics(): void {
   });
 
   featureLibsFolders.forEach((lib) =>
-    execSync(`yarn --cwd feature-libs/${lib} run test:schematics`, {
+    execSync(`yarn --cwd feature-libs/${lib} run test:schematics --coverage`, {
```
Comment (marlass) [inline tools/schematics/testing.ts:132]: Shouldn't we also add `--coverage` for the integration libs loop?

**Review by marlass — CHANGES_REQUESTED** [review]
Could you also change `.release-it` config for cds to run the build:cds script instead of ng build?

Also add the CDS schematics tests to CI unit tests script

File: projects/schematics/src/shared/utils/lib-utils.ts (hunk 1 above)
Comment (znikola) [reply projects/schematics/src/shared/utils/lib-utils.ts:89]: I actually like it, as it simplifies usage for cases when we specify only one custom config.
It's also aligned with `addModuleImport()` from `projects/schematics/src/shared/utils/new-module-utils.ts`.

And it doesn't really bring much complexity, IMO, as we just use the `[].concat()` trick.

**Review by marlass — APPROVED** [review]

**Pull request merged**